compiler/src/compile.rs

```diff
@@ -1256,66 +1248,120 @@ impl<O: OutputStream> Compiler<O> {
         self.emit(Instruction::BinaryOperation { op: i, inplace });
     }
 
-    fn compile_test(
+    /// Implement boolean short circuit evaluation logic.
+    /// https://en.wikipedia.org/wiki/Short-circuit_evaluation
+    ///
+    /// This means, in a boolean statement 'x and y' the variable y will
+    /// not be evaluated when x is false.
+    ///
+    /// The idea is to jump to a label if the expression is either true or false
+    /// (indicated by the condition parameter).
+    fn compile_jump_if(
         &mut self,
         expression: &ast::Expression,
-        true_label: Option<Label>,
-        false_label: Option<Label>,
-        context: EvalContext,
+        condition: bool,
```

Comment:
I'd recommend moving `condition` to the first argument and removing the second underscore in the function name so it can read better, e.g. you can see `compile_jumpif(true, ...)` compiles a `JumpIfTrue`.

Reply:
The way I intended this to read was: `compile_jump_if(expression, true, target)` means: Jump if the expression is true to target.

Reply:
I'll leave this for now like it is. Thanks for the idea!